server.go

```diff
@@ -121,3 +154,48 @@ func runHTTP(addr, path string, handler http.Handler, landing []byte) {
 	log.Infof("Starting HTTP server for http://%s%s ...", addr, path)
 	log.Fatal(srv.ListenAndServe())
 }
+
+func runHTTPSfunc(addr, path string, handler func(http.ResponseWriter, *http.Request), landing []byte) {
+	mux := http.NewServeMux()
+	mux.HandleFunc(path, handler)
+	mux.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
+		w.Header().Add("Strict-Transport-Security", "max-age=63072000; includeSubDomains")
+		w.Write(landing)
+	})
+
+	tlsCfg := &tls.Config{
+		MinVersion:               tls.VersionTLS12,
+		CurvePreferences:         []tls.CurveID{tls.CurveP521, tls.CurveP384, tls.CurveP256},
+		PreferServerCipherSuites: true,
+		CipherSuites: []uint16{
+			tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+			tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
+			tls.TLS_RSA_WITH_AES_256_GCM_SHA384,
+			tls.TLS_RSA_WITH_AES_256_CBC_SHA,
+		},
+	}
+
+	srv := &http.Server{
+		Addr:         addr,
+		Handler:      mux,
+		TLSConfig:    tlsCfg,
+		TLSNextProto: make(map[string]func(*http.Server, *tls.Conn, http.Handler)), // disable HTTP/2
+	}
+	log.Infof("Starting HTTPS server for https://%s%s ...", addr, path)
+	log.Fatal(srv.ListenAndServeTLS(*sslCertFileFKingpin, *sslKeyFileFKingpin))
+}
+
+func runHTTPfunc(addr, path string, handler func(http.ResponseWriter, *http.Request), landing []byte) {
+	mux := http.NewServeMux()
+	mux.HandleFunc(path, handler)
+	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Write(landing)
+	})
+
+	srv := &http.Server{
+		Addr:    addr,
+		Handler: mux,
+	}
+	log.Infof("Starting HTTP server for http://%s%s ...", addr, path)
+	log.Fatal(srv.ListenAndServe())
+}
```

Comment:
I'm not sure why we need those 3 extra functions (why?), but if we do, they should not be copy&pasted.

Reply:
Functions have been removed